Source/WebCore/html/FTPDirectoryDocument.cpp

```diff
@@ -67,14 +66,12 @@ class FTPDirectoryDocumentParser final : public HTMLDocumentParser {
     void append(RefPtr<StringImpl>&&) override;
     void finish() override;
 
-    void checkBuffer(int len = 10)
+    void checkBuffer(size_t len = 10)
     {
-        if ((m_dest - m_buffer) > m_size - len) {
+        if (m_destIndex > (m_size - len)) {
             // Enlarge buffer
             int newSize = std::max(m_size * 2, m_size + len);
-            int oldOffset = m_dest - m_buffer;
-            m_buffer = static_cast<UChar*>(fastRealloc(m_buffer, newSize * sizeof(UChar)));
-            m_dest = m_buffer + oldOffset;
+            m_buffer.realloc(newSize * sizeof(UChar));
```

Comment:
Seems like this could overflow.

Reply:
Used `CheckedSize` to guard against overflows.